File: app/src/main/java/com/battlelancer/seriesguide/extensions/AvailableExtensionsLoader.java
package com.battlelancer.seriesguide.extensions;

import android.content.Context;
import androidx.annotation.NonNull;
import com.uwetrottmann.androidutils.GenericSimpleLoader;
import java.util.List;

/**
 * Queries for any installed {@link com.battlelancer.seriesguide.api.SeriesGuideExtension}
 * extensions.
 */
class AvailableExtensionsLoader extends GenericSimpleLoader<List<Extension>> {

    AvailableExtensionsLoader(Context context) {
        super(context);
    }

    @NonNull
    @Override
    public List<Extension> loadInBackground() {
        return ExtensionManager.get(getContext()).queryAllAvailableExtensions(getContext());
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/extensions/AvailableExtensionsLoader.kt
// Copyright 2023 Uwe Trottmann
// SPDX-License-Identifier: Apache-2.0

package com.battlelancer.seriesguide.extensions

import android.content.Context
import com.uwetrottmann.androidutils.GenericSimpleLoader

/**
 * Queries for any installed [com.battlelancer.seriesguide.api.SeriesGuideExtension]
 * extensions.
 */
class AvailableExtensionsLoader(context: Context) :
    GenericSimpleLoader<MutableList<Extension>>(context) {

    override fun loadInBackground(): MutableList<Extension> {
        return ExtensionManager.get(context).queryAllAvailableExtensions(context)
    }

}